benchmarks/bench_verkle_primitives.nim

```diff
@@ -0,0 +1,144 @@
+export notes
+
+type
+  Prj* = ECP_TwEdwards_Prj[Fp[Banderwagon]]
+  Aff* = ECP_TwEdwards_Prj[Fp[Banderwagon]]
+
+const Iters = 10000
+
+proc separator*() = separator(152)
+
+proc report(op, domain: string, start, stop: MonoTime, startClk, stopClk: int64, iters: int) =
+  let ns = inNanoseconds((stop-start) div iters)
+  let throughput = 1e9 / float64(ns)
+  when SupportsGetTicks:
+    echo &"{op:<35} {domain:<40} {throughput:>15.3f} ops/s     {ns:>9} ns/op     {(stopClk - startClk) div iters:>9} CPU cycles (approx)"
+  else:
+    echo &"{op:<35} {domain:<40} {throughput:>15.3f} ops/s     {ns:>9} ns/op"
+
+macro fixEllipticDisplay(T: typedesc): untyped =
+  # At compile-time, enums are integers and their display is buggy
+  # we get the Curve ID instead of the curve name.
+  let instantiated = T.getTypeInst()
+  var name = $instantiated[1][0] # EllipticEquationFormCoordinates
+  let fieldName = $instantiated[1][1][0]
+  let curveName = $Curve(instantiated[1][1][1].intVal)
+  name.add "[" & fieldName & "[" & curveName & "]]"
+  result = newLit name
+
+macro fixFieldDisplay(T: typedesc): untyped =
+  # At compile-time, enums are integers and their display is buggy
+  # we get the Curve ID instead of the curve name.
+  let instantiated = T.getTypeInst()
+  var name = $instantiated[1][0] # Fp
+  name.add "[" & $Curve(instantiated[1][1].intVal) & "]"
+  result = newLit name
+
+func fixDisplay(T: typedesc): string =
+  when T is (ECP_ShortW_Prj or ECP_ShortW_Jac or ECP_ShortW_Aff):
+    fixEllipticDisplay(T)
+  elif T is (Fp or Fp2 or Fp4 or Fp6 or Fp12):
+    fixFieldDisplay(T)
+  else:
+    $T
+
+func fixDisplay(T: Curve): string =
+  $T
+
+template bench(op: string, T: typed, iters: int, body: untyped): untyped =
+  measure(iters, startTime, stopTime, startClk, stopClk, body)
+  report(op, fixDisplay(T), startTime, stopTime, startClk, stopClk, iters)
+
+# proc equalityBench*(T: typedesc, iters: int) =
+#   when T is Aff:
+#     let P = Banderwagon.getGenerator()
+#     let Q = Banderwagon.getGenerator()
+#   else:
+#     var P, Q: Prj
+#     P.fromAffine(Banderwagon.getGenerator())
+#     Q.fromAffine(Banderwagon.getGenerator())
+#   bench("Banderwagon Equality ", T, iters):
+#     assert (P == Q).bool()
+
+
+
+proc serializeBench*(T: typedesc, iters: int) =
+  var bytes: array[32, byte]
+  var P: Prj
+  P.fromAffine(Banderwagon.getGenerator())
+  for i in 0 ..< 9:
+    P.double()
+  bench("Banderwagon Serialization", T, iters):
+    discard bytes.serialize(P)
+
+proc deserializeBench*(T: typedesc, iters: int) =
+  var bytes: array[32, byte]
+  var P: Prj
+  P.fromAffine(Banderwagon.getGenerator())
+  for i in 0 ..< 6:
+    P.double()
+  discard bytes.serialize(P)
+  bench("Banderwagon Deserialization", T, iters):
+    discard P.deserialize(bytes)
+
+
+proc main() =
+  # separator()
+  # equalityBench(Aff, Iters)
+  # equalityBench(Prj, Iters)
+  separator()
+  serializeBench(Prj, Iters)
+  deserializeBench(Prj, Iters)
```

Comment:
We probably need `deserialize_vartime` and `deserialize_unchecked_vartime` benches now